example/hooks/useCanonicalUrl.ts

```diff
@@ -0,0 +1,13 @@
+import { useRouter } from 'next/router';
+import { getBasePath } from '../../lib/helpers/getBasePath';
+import { getOrigin } from '../../lib/helpers/getOrigin';
+
+export const useCanonicalUrl = (locale: string): string | null => {
+  const { asPath, basePath } = useRouter();
+  const [path, query] = asPath.split('?');
+  const origin = getOrigin();
+  // new URLSearchParams will make sure that any special character will be parsed
+  const queryString = new URLSearchParams(query).toString();
+
+  return `${origin}${getBasePath(basePath)}${locale}${path}${queryString ? '?' + queryString : ''}`;
+};
```

Comment:
Yes you are right. I though about exporting the helpers bit it makes no sense, I wanted to avoid duplication. What do you suggest? Have useCanonicalUrl as part of the package and export it? Or simply not depend on the helpers? The thing is I'd like to avoid duplication.

Reply:
It's good to keep in mind that there's duplication, and then there's duplication. Within either the npm-published module or the example code, duplication isn't a good idea. For functionality that's implemented separately in both, duplication is a much smaller cost compared to a messy API.